src/Entities/Transaction/TransactionItemPreviewWithPrice.php

```diff
@@ -16,7 +16,7 @@
 class TransactionItemPreviewWithPrice
 {
     private function __construct(
-        public Price $price,
+        public Price|TransactionPreviewPrice $price,
```

Comment:
We could make this `TransactionPreviewPrice` only?

Reply:
Just `TransactionPreviewPrice` would make it breaking so lets stick with the union 👍